### PR TITLE
feature/ads 29 approving a pet for adoption us12

### PR DESCRIPTION
**Pull request by trpubz**
As a visitor
When I visit an admin application show page ('/admin/applications/:id')
For every pet that the application is for, I see a button to approve the application for that specific pet
When I click that button
Then I'm taken back to the admin application show page
And next to the pet that I approved, I do not see a button to approve this pet
And instead I see an indicator next to the pet that they have been approved

### CODE REVIEW

File: spec/features/admin_applications/show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    # When I click that button
+    # Then I'm taken back to the admin application show page
+    # And next to the pet that I approved, I do not see a button to approve this pet
+    # And instead I see an indicator next to the pet that they have been approved
+    visit "/admin/applications/#{@app2.id}"
+
+    within("#pet-#{@p1.id}") do
+      click_button "Approve"
+    end
+
```
Comment (chuckrenny) [inline spec/features/admin_applications/show_spec.rb:38]: between the two within blocks i'd add a line to check if the current route did take the user back to admin application show page based on line 30

File: app/views/admin_applications/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+<h4>Pets</h4>
+<% @app.pets.each do | pet | %>
+    <section id="pet-<%= pet.id %>">
+      <span><%= pet.name %></span>
+      <% if pet.adoptable %>
+          <%= form_with url: "/pets/#{pet.id}?approve=true&app_id=#{@app.id}",
+                        method: :patch,
+                        data: {turbo: false},
+                        html: {style: "display: inline-block"} do |form| %>
+            <% form.submit "Approve"  %>
```
Comment (chuckrenny) [inline app/views/admin_applications/show.html.erb:16]: Is the html block a sort of form formatting? I haven't seen that before

File: app/controllers/pets_controller.rb [hunk 3]
```diff
@@ -31,9 +31,15 @@ def edit
 
   def update
     pet = Pet.find(params[:id])
-    if pet.update(pet_params)
+    if params[:approve]
+      # approve == true ==== adoptable == false
+      pet.update(adoptable: !ActiveRecord::Type::Boolean.new.cast(params[:approve]))
+      pet.save
```
Comment (chuckrenny) [inline app/controllers/pets_controller.rb:37]: What led to the ActiveRecord type match in line 36?

File: app/controllers/admin_applications_controller.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class AdminApplicationsController < ApplicationController
```
Comment (chuckrenny) [inline app/controllers/admin_applications_controller.rb:1]: AdminApplications works as a name but it makes it seem there's an Admin table and this is a joins controller between Admin and Application tables. Admin may be more clear here

**Comment by chuckrenny** [comment]
To complete US 12 in a way to setup US 13&14, we need a AR method to change a specific application status on a pet.

File: app/views/admin_applications/show.html.erb (hunk 2 above)
Comment (trpubz) [reply app/views/admin_applications/show.html.erb:16]: yeah, I wanted everything inline, and stack overflow, master sensei, taught me this one.  without the span tag and the style inline-block then everything would've been stacked vertically

File: app/controllers/admin_applications_controller.rb (hunk 4 above)
Comment (trpubz) [reply app/controllers/admin_applications_controller.rb:1]: This is a unique functionality, and though not a joins function, it has it's own unique show view, different from ApplicationsController.

Comment (chuckrenny) [reply app/controllers/admin_applications_controller.rb:1]: I would just rename it AdminController. ATP it comes down to preference like most things in code

**Comment by chuckrenny** [comment]
New code satisfies the US and sets us up for the next two. Raising an exception is a nice addition

**Pull request merged**